### PR TITLE
Exclude queued experiments from select experiments for plots quick pick

### PR DESCRIPTION
**Pull request by mattseddon**
### Demo

https://user-images.githubusercontent.com/37993418/190561301-68ac36fe-ef4f-43f1-a696-e3d84ac7c314.mov

### CODE REVIEW

**Review by julieg18 — APPROVED** [review]
Looks good!

**Review by sroy3 — APPROVED** [review]

File: extension/src/experiments/model/index.ts [hunk 1]
```diff
@@ -324,13 +324,15 @@ export class ExperimentsModel extends ModelWithPersistence {
...
-    })
+    return this.getExperiments()
+      .filter(({ queued }) => !queued)
+      .map(experiment => {
+        const checkpoints = this.getCheckpointsWithType(experiment.id)
+        if (!definedAndNonEmpty(checkpoints)) {
+          return experiment
+        }
+        return { ...experiment, checkpoints }
+      })
```
Comment (wolmir) [inline extension/src/experiments/model/index.ts:335]: What about the following, to get rid of the filter and map?
```
  public getExperimentsWithCheckpoints(): ExperimentWithCheckpoints[] {
    const experimentsWithCheckpoints: ExperimentWithCheckpoints[] = []
    const experiments = this.getExperiments()

    for (const experiment of experiments) {
      const { queued, id } = experiment
      const checkpoints = !queued && this.getCheckpointsWithType(id)

      if (definedAndNonEmpty(checkpoints)) {
        experimentsWithCheckpoints.push({
          ...experiment,
          checkpoints
        })
      }
    }

    return experimentsWithCheckpoints
  }
```

And in `array.ts`:
```
export const definedAndNonEmpty = (
  maybeArray: readonly unknown[] | undefined | false
): maybeArray is unknown[] => {
  return !!(maybeArray && maybeArray.length > 0)
}
```

Comment (mattseddon) [reply extension/src/experiments/model/index.ts:335]: I think 
```
      const checkpoints = !queued && this.getCheckpointsWithType(id)

      if (definedAndNonEmpty(checkpoints)) {
```
needs to be 

```      
      if (queued){
        continue
      }
      const checkpoints = this.getCheckpointsWithType(id)

      if (!definedAndNonEmpty(checkpoints)) {
        experimentsWithCheckpoints.push(experiment)
        continue
      }

      ...
```

Comment (mattseddon) [reply extension/src/experiments/model/index.ts:335]: I'll update to simplify 👍🏻.

**Pull request merged**